test/project_types/script/layers/application/extension_points_test.rb

```diff
@@ -68,6 +68,68 @@
     end
   end
 
+  describe ".all_languages" do
+    subject { Script::Layers::Application::ExtensionPoints.all_languages }
+
+    let(:scripts_beta_extension_points) { false }
+    let(:scripts_beta_languages) { false }
+
+    before do
+      ShopifyCLI::Feature
+        .expects(:enabled?)
+        .with(:scripts_beta_languages)
+        .returns(scripts_beta_languages)
+        .at_least_once
+      ShopifyCLI::Feature
+        .expects(:enabled?)
+        .with(:scripts_beta_extension_points)
+        .returns(scripts_beta_extension_points)
+        .at_least_once
+    end
+
+    describe "when beta language flag is enabled" do
+      let(:scripts_beta_languages) { true }
+
+      it "returns a list of all languages implemented by non beta extension points" do
+        assert_equal 2, subject.count
+        assert_equal "assemblyscript", subject[0]
+        assert_equal "rust", subject[1]
```

Comment:
Nit: I think this would read nicely as:

```ruby
assert_equal(["assemblyscript", "rust"], subject)
```

Also has the benefit more verbose message in case there's a mismatch in the number of elements in the array.